src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.js

```diff
@@ -0,0 +1,153 @@
+import React, {
+  memo,
+  useRef,
+  useState,
+  useLayoutEffect,
+} from 'react';
+import { PropTypes } from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import { get } from 'lodash';
+
+import {
+  createOkapiHeaders,
+  createUrl,
+  isFormattedMessage,
+  isTranslationId,
+} from '../../../../utils';
+
+import { OptionsList } from '../partials';
+
+import styles from './withReferenceValues.css';
+
+export const withReferenceValues = memo(props => {
+  const {
+    id,
+    input,
+    WrappedComponent,
+    wrapperLabel,
+    wrapperSourceLink,
+    wrapperSourcePath,
+    okapi,
+    ...rest
+  } = props;
+
+  const [hasLoaded, setHasLoaded] = useState(false);
+  const [dataOptions, setDataOptions] = useState([]);
+  const [currentValue, setCurrentValue] = useState(input?.value || null);
+  const [wrapperValue, setWrapperValue] = useState(null);
+
+  const fetchList = async () => {
+    try {
+      const response = await fetch(
+        createUrl(`${okapi.url}${wrapperSourceLink}`, null, false),
+        { headers: { ...createOkapiHeaders(okapi) } },
+      );
+      const body = await response.json();
+      const curDataOptions = get(body, wrapperSourcePath, []);
+
+      setDataOptions(curDataOptions);
+      setHasLoaded(true);
+    } catch (e) {
+      console.log('Error: ', e); // eslint-disable-line no-console
+      setHasLoaded(false);
+    }
+  };
+
+  useLayoutEffect(() => {
+    let setCancel = false;
+
+    if (!setCancel) {
+      fetchList().then();
+    }
+
+    return () => {
+      setCancel = true;
+    };
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useLayoutEffect(() => {
+    let newValue = '';
+
+    if (currentValue && wrapperValue) {
+      newValue = `${currentValue} ${wrapperValue}`;
+    } else if (!currentValue && wrapperValue) {
+      newValue = wrapperValue;
+    }
+
+    setCurrentValue(newValue);
+  }, [wrapperValue]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const handleChange = e => {
+    setCurrentValue(e.target.value);
+    input.onChange(e);
```

Comment:
Consider using object destructuring assignment.
```suggestion
    const {
        onChange,
        onBlur,
      ...
    } = input;
```